### PR TITLE
fix redoc build and cleanup warnings

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** #165

**Proposed Changes:**

1. change redoc build to use openapi-cli instead of swagger-cli, which should fix the issue with the published version of 
2. replace index.html with current recommended content from ReDoc
2. cleanup redoc warnings

**PR Checklist:**

- [X] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [X] This PR has **no** breaking changes.
- [X] This PR does not make any changes to the core spec in the `stac-spec` directory (these are included as a subtree and should be updated directly in [radiantearth/stac-spec](https://github.com/radiantearth/stac-spec))
- [X] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-api-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "repository": "https://github.com/radiantearth/stac-api-spec",
   "license": "Apache-2.0",
   "scripts": {
-    "serve": "redoc-cli serve core/openapi.yaml --watch --options.expandResponses \"200,201,202,203,204\" --options.pathInMiddlePanel true",
+    "serve": "redoc-cli serve $npm_config_class/openapi.yaml --watch --options.expandResponses \"200,201,202,203,204\" --options.pathInMiddlePanel true",
```
Comment (philvarner) [inline package.json:8]: syntax is now `npm run server --class=core` (where class can be core, ogcapi-features, item-search)

**Review by matthewhanson — APPROVED** [review]

File: build/index.html [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
-   <meta http-equiv="x-ua-compatible" content="ie=edge">
-   <link rel="shortcut icon" href="../assets/images/favicon.png">
-   <title>STAC API</title>
- </head>
- <body>
-   <div id="content"></div>
-   <script src="https://cdn.jsdelivr.net/npm/redoc@next/bundles/redoc.standalone.js"> </script>
-   <script type="text/javascript">
-     Redoc.init('openapi.yaml', {
-       expandResponses:"200,201,202,203,204",
```
Comment (m-mohr) [inline build/index.html:14]: Is it intentional that you drop those settings?

File: package.json (hunk 1 above)
Comment (m-mohr) [reply package.json:8]: Does that work on all major OS (looking at Windows)?

Comment (philvarner) [reply package.json:8]: It's part of NPM rather than shell, so it should.

File: build/index.html (hunk 2 above)
Comment (philvarner) [reply build/index.html:14]: I just readded them. I'm not sure why we made these changes to the default, but I also didn't test them to see how they were different. I'll try that out before changing.

Comment (m-mohr) [reply build/index.html:14]: I made them, they give a bit better UI experience IMHO.

**Pull request merged**